genai-perf/genai_perf/profile_data_parser/image_retrieval_profile_data_parser.py

```diff
@@ -76,9 +82,26 @@ def _parse_requests(self, requests: dict) -> ImageRetrievalMetrics:
         benchmark_duration = (max_res_timestamp - min_req_timestamp) / 1e9  # to seconds
         request_throughputs = [len(requests) / benchmark_duration]
 
-        return ImageRetrievalMetrics(
+        self._image_metric = ImageRetrievalMetrics(
             request_throughputs,
             request_latencies,
             image_throughputs,
             image_latencies,
         )
+
+        self._calculate_goodput(benchmark_duration)
+
+        return self._image_metric
+
+    def _calculate_goodput(self, benchmark_duration) -> None:
+        if self._goodput_constraints:
+            llm_goodput_calculator = LLMGoodputCalculator(
+                self._goodput_constraints,
+                self._image_metric,
+                benchmark_duration,
+            )
+
+            llm_goodput_calculator.compute()
+            self._image_metric.request_goodputs = llm_goodput_calculator.goodput
+        else:
+            return
```

Comment:
all `ProfileDataParser`, `LLMProfileDataParser`, and `ImageRetrievalProfileDataParser` shares this identical code block. let's only implement this in the base class and let all the derived classes use that one instead of having their own copy.

Reply:
@dyastremsky making this change, I believe, will address your question.

Reply:
@dyastremsky @nv-hwoo See below whether the new iteration addresses your question. 
Updated according to all suggestions in this round.